### PR TITLE
Nicer zsh completion

### PR DESCRIPTION
**Pull request by mkoskar**
This is completion based on ``zshcompsys`` rather than older ``zshcompctl``.

### CODE REVIEW

File: completions/pyenv.zsh [hunk 1]
```diff
@@ -1,18 +1,12 @@
-if [[ ! -o interactive ]]; then
```
Comment (mayanksuman) [inline completions/pyenv.zsh]: Why this is removed?

The completion should not be loaded if zsh is not interactive. Please revert this change.

Comment (mkoskar) [reply completions/pyenv.zsh]: The whole point is that you just drop the file in your ``fpath`` and you're done. That's how majority of ``zsh`` completions are done and it's also suitable for packagers. Alternative is to ``compdef`` it yourself if you prefer and that you can do from your ``.zshrc`` after "interactive check".

Comment (mayanksuman) [reply completions/pyenv.zsh]: Yeah! It is easier only when a majority of users get pyenv by packaging, but most of them get it by git clone. The `pyenv-installer` also uses `git clone`.

The completion file in question is dynamically loaded when user run `eval $(pyenv init -)` in `~/.zshrc`. However, if want to add it to `fpath`, it is not as easy as sourcing. Please refer to [this answer]( https://unix.stackexchange.com/a/213658).

Comment (mayanksuman) [reply completions/pyenv.zsh]: Further, I guess the output of `pyenv init -` need to change for the changes in completion file.

Comment (mkoskar) [reply completions/pyenv.zsh]: Ah yeah, I see ... I haven't noticed it in my installation. Indeed running from git working copy include ``source .../pyenv.zsh``. TBH I don't run ``eval $(pyenv init -)`` as I don't need all the bell and whistles. This was inspired by Archlinux' ``pyenv`` package which includes ``pyenv.zsh`` in ``/usr/share/zsh/site-functions`` and obviously doesn't work, even doing ``eval $(pyenv init -)`` wouldn't help.

Not sure where to get from here, as for me I'm happy to keep this in my ``~/.zfunctions/_pyenv`` but tweaking ``libexec/pyenv-init`` a bit might solve it for both usecases.

**Comment by adigitoleo** [comment]
Either this issue or #1644 could be closed to avoid confusing potential contributors.
